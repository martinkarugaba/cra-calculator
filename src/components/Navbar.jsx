import { NavLink } from 'react-router-dom';

const Navbar = () => (
  <div className="flex justify-between items-center px-16 py-6">
    <div>
      <h1 className="text-4xl font-medium text-slate-600">Math Magicians</h1>
    </div>

    <ul className="flex justify-center items-center gap-x-4 text-teal-700 font-medium">
      <li className="border-r-2 border-gray-950 pr-4 hover:text-slate-600 ">
        <NavLink to="/">Home</NavLink>
      </li>
      <li className="border-r-2 cursor-pointer border-gray-950 pr-4 hover:text-slate-600 ">
        <NavLink to="calculator">Calculator</NavLink>
      </li>
      <li className="border-gray-950 cursor-pointer pr-4 hover:text-slate-600">
        <NavLink to="quote">Quote</NavLink>
      </li>
    </ul>
  </div>
);
export default Navbar;
