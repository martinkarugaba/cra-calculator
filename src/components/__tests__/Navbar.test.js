import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../Navbar';

describe('Navbar component', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders the navigation links correctly', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );

    const homeLink = getByText('Home');
    const calculatorLink = getByText('Calculator');
    const quoteLink = getByText('Quote');

    expect(homeLink).toBeInTheDocument();
    expect(calculatorLink).toBeInTheDocument();
    expect(quoteLink).toBeInTheDocument();

    expect(homeLink.getAttribute('href')).toBe('/');
    expect(calculatorLink.getAttribute('href')).toBe('/calculator');
    expect(quoteLink.getAttribute('href')).toBe('/quote');
  });
});
