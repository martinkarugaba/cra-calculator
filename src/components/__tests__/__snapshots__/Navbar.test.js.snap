// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component matches the snapshot 1`] = `
<div>
  <div
    class="flex justify-between items-center px-16 py-6"
  >
    <div>
      <h1
        class="text-4xl font-medium text-slate-600"
      >
        Math Magicians
      </h1>
    </div>
    <ul
      class="flex justify-center items-center gap-x-4 text-teal-700 font-medium"
    >
      <li
        class="border-r-2 border-gray-950 pr-4 hover:text-slate-600 "
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="border-r-2 cursor-pointer border-gray-950 pr-4 hover:text-slate-600 "
      >
        <a
          class=""
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li
        class="border-gray-950 cursor-pointer pr-4 hover:text-slate-600"
      >
        <a
          class=""
          href="/quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </div>
</div>
`;
